src/features/terminal/terminalEnvVarInjector.ts

```diff
@@ -109,15 +110,45 @@ export class TerminalEnvVarInjector implements Disposable {
     private async injectEnvironmentVariablesForWorkspace(workspaceFolder: WorkspaceFolder): Promise<void> {
         const workspaceUri = workspaceFolder.uri;
         try {
-            const envVars = await this.envVarManager.getEnvironmentVariables(workspaceUri);
-
+            // Check if environment variable injection is enabled
+            const config = getConfiguration('python', workspaceUri);
+            const useEnvFile = config.get<boolean>('terminal.useEnvFile', false);
+            const envFilePath = config.get<string>('envFile');
+            
             // use scoped environment variable collection
             const envVarScope = this.getEnvironmentVariableCollectionScoped({ workspaceFolder });
             envVarScope.clear(); // Clear existing variables for this workspace
 
+            // Check if python.envFile is set but useEnvFile is false (default) and show notification
+            if (!useEnvFile && envFilePath) {
+                traceVerbose(
+                    `TerminalEnvVarInjector: python.envFile is set but python.terminal.useEnvFile is false for workspace: ${workspaceUri.fsPath}`,
+                );
+                
+                // Show information message to user
+                showInformationMessage(
+                    'The python.envFile setting is configured but will not take effect in terminals. Enable the "python.terminal.useEnvFile" setting to use environment variables from .env files in terminals.',
+                    'Open Settings'
+                ).then((selection) => {
+                    if (selection === 'Open Settings') {
+                        // Open VS Code settings to the python.terminal.useEnvFile setting
+                        import('vscode').then(vscode => {
+                            vscode.commands.executeCommand('workbench.action.openSettings', 'python.terminal.useEnvFile');
+                        });
+                    }
+                });
+            }
+
+            if (!useEnvFile) {
+                traceVerbose(
+                    `TerminalEnvVarInjector: Environment variable injection disabled for workspace: ${workspaceUri.fsPath}`,
```

Comment:
Switch this to only inject if useEnvFile is true- the turn to get out of injecting is a weird way to do this